### ISSUE TITLE
Audit of retained files

### ISSUE THREAD

**Issue by Th3Van** [storj/storj#4134]
_**SN ID : 12q8TgMpvw3oJT3uLhfoXo4sN6Tg5mDtwTxT48RvLowz5cttgct**_
```
root@server030:/disk102/storj/logs/storagenode# grep "T3LQD7RI6JDRRN5IHSIANELEVD2S6ZQEGIMR4EX7WO4GHSU34R3Q" ./old/server013* server013-v1.*
server013-v1.24.4-30013.log:2021-03-07T03:35:16.931+0100  	INFO    piecestore      upload started  	{"Piece ID": "T3LQD7RI6JDRRN5IHSIANELEVD2S6ZQEGIMR4EX7WO4GHSU34R3Q", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "PUT", "Available Space": 10632486993280}
server013-v1.24.4-30013.log:2021-03-07T03:35:20.201+0100  	INFO    piecestore      uploaded        	{"Piece ID": "T3LQD7RI6JDRRN5IHSIANELEVD2S6ZQEGIMR4EX7WO4GHSU34R3Q", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "PUT", "Size": 2212096}

server013-v1.30.2-30013.log:2021-06-01T17:11:07.369+0200        DEBUG   retain  About to move piece to trash    {"Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Piece ID": "T3LQD7RI6JDRRN5IHSIANELEVD2S6ZQEGIMR4EX7WO4GHSU34R3Q", "Status": "enabled"}

server013-v1.30.2-30013.log:2021-06-02T20:33:29.529+0200        INFO    piecestore      download started        {"Piece ID": "T3LQD7RI6JDRRN5IHSIANELEVD2S6ZQEGIMR4EX7WO4GHSU34R3Q", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "GET_AUDIT"}
server013-v1.30.2-30013.log:2021-06-02T20:33:29.530+0200        ERROR   piecestore      download failed 	{"Piece ID": "T3LQD7RI6JDRRN5IHSIANELEVD2S6ZQEGIMR4EX7WO4GHSU34R3Q", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "GET_AUDIT", "error": "file does not exist", "errorVerbose": "file does not exist\n\tstorj.io/common/rpc/rpcstatus.Wrap:74\n\tstorj.io/storj/storagenode/piecestore.(*Endpoint).Download:508\n\tstorj.io/common/pb.DRPCPiecestoreDescription.Method.func2:217\n\tstorj.io/drpc/drpcmux.(*Mux).HandleRPC:33\n\tstorj.io/common/rpc/rpctracing.(*Handler).HandleRPC:58\n\tstorj.io/drpc/drpcserver.(*Server).handleRPC:111\n\tstorj.io/drpc/drpcserver.(*Server).ServeOne:62\n\tstorj.io/drpc/drpcserver.(*Server).Serve.func2:99\n\tstorj.io/drpc/drpcctx.(*Tracker).track:51"}
```

-------------------

_**SN ID : 1NfkK7vAEdJkPsZBqXkHFxbDvqN1aHZZuV5mbYEU1ZSCePr1ct**_

```
root@server030:/disk102/storj/logs/storagenode# grep "JGYTRNRQVAXSRYHBWYZM7UBGO3OE5Q3KGIML64HCSKPJKBMAKN5Q" ./old/server019*.log server019*.log
server019-v1.25.2-30019.log:2021-03-16T13:52:51.757+0100  	INFO    piecestore      upload started  {"Piece ID": "JGYTRNRQVAXSRYHBWYZM7UBGO3OE5Q3KGIML64HCSKPJKBMAKN5Q", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "PUT", "Available Space": 10522546847232}
server019-v1.25.2-30019.log:2021-03-16T13:52:55.341+0100  	INFO    piecestore      uploaded        {"Piece ID": "JGYTRNRQVAXSRYHBWYZM7UBGO3OE5Q3KGIML64HCSKPJKBMAKN5Q", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "PUT", "Size": 2212096}

server019-v1.30.2-30019.log:2021-06-01T17:26:23.319+0200        DEBUG   retain  About to move piece to trash    {"Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Piece ID": "JGYTRNRQVAXSRYHBWYZM7UBGO3OE5Q3KGIML64HCSKPJKBMAKN5Q", "Status": "enabled"}

server019-v1.30.2-30019.log:2021-06-02T23:52:16.062+0200        INFO    piecestore      download started        {"Piece ID": "JGYTRNRQVAXSRYHBWYZM7UBGO3OE5Q3KGIML64HCSKPJKBMAKN5Q", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "GET_AUDIT"}
server019-v1.30.2-30019.log:2021-06-02T23:52:16.063+0200        ERROR   piecestore      download failed {"Piece ID": "JGYTRNRQVAXSRYHBWYZM7UBGO3OE5Q3KGIML64HCSKPJKBMAKN5Q", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "GET_AUDIT", "error": "file does not exist", "errorVerbose": "file does not exist\n\tstorj.io/common/rpc/rpcstatus.Wrap:74\n\tstorj.io/storj/storagenode/piecestore.(*Endpoint).Download:508\n\tstorj.io/common/pb.DRPCPiecestoreDescription.Method.func2:217\n\tstorj.io/drpc/drpcmux.(*Mux).HandleRPC:33\n\tstorj.io/common/rpc/rpctracing.(*Handler).HandleRPC:58\n\tstorj.io/drpc/drpcserver.(*Server).handleRPC:111\n\tstorj.io/drpc/drpcserver.(*Server).ServeOne:62\n\tstorj.io/drpc/drpcserver.(*Server).Serve.func2:99\n\tstorj.io/drpc/drpcctx.(*Tracker).track:51"}
```

-------------------

_**SN ID : 15owzZ7986YEtfz9etw86Yw3GwSRd6zQFtfUDwChFvRrKPJL5a**_
```
root@server030:/disk102/storj/logs/storagenode# grep "A5QSWO4LN4E4URA3E5T7L4B3UV5JIN6HSLDKKXRB6UA57HFYFOIQ" ./old/server029* server029-v1.*
server029-v1.25.2-30029.log:2021-03-14T07:34:05.478+0100  	INFO    piecestore      upload started  {"Piece ID": "A5QSWO4LN4E4URA3E5T7L4B3UV5JIN6HSLDKKXRB6UA57HFYFOIQ", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "PUT", "Available Space": 10619900493312}
server029-v1.25.2-30029.log:2021-03-14T07:34:07.701+0100  	INFO    piecestore      uploaded        {"Piece ID": "A5QSWO4LN4E4URA3E5T7L4B3UV5JIN6HSLDKKXRB6UA57HFYFOIQ", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "PUT", "Size": 2212096}

server029-v1.30.2-30029.log:2021-06-01T11:15:57.731+0200        DEBUG   retain  About to move piece to trash    {"Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Piece ID": "A5QSWO4LN4E4URA3E5T7L4B3UV5JIN6HSLDKKXRB6UA57HFYFOIQ", "Status": "enabled"}

server029-v1.30.2-30029.log:2021-06-02T20:52:24.517+0200        INFO    piecestore      download started        {"Piece ID": "A5QSWO4LN4E4URA3E5T7L4B3UV5JIN6HSLDKKXRB6UA57HFYFOIQ", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "GET_AUDIT"}
server029-v1.30.2-30029.log:2021-06-02T20:52:24.518+0200        ERROR   piecestore      download failed {"Piece ID": "A5QSWO4LN4E4URA3E5T7L4B3UV5JIN6HSLDKKXRB6UA57HFYFOIQ", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "GET_AUDIT", "error": "file does not exist", "errorVerbose": "file does not exist\n\tstorj.io/common/rpc/rpcstatus.Wrap:74\n\tstorj.io/storj/storagenode/piecestore.(*Endpoint).Download:508\n\tstorj.io/common/pb.DRPCPiecestoreDescription.Method.func2:217\n\tstorj.io/drpc/drpcmux.(*Mux).HandleRPC:33\n\tstorj.io/common/rpc/rpctracing.(*Handler).HandleRPC:58\n\tstorj.io/drpc/drpcserver.(*Server).handleRPC:111\n\tstorj.io/drpc/drpcserver.(*Server).ServeOne:62\n\tstorj.io/drpc/drpcserver.(*Server).Serve.func2:99\n\tstorj.io/drpc/drpcctx.(*Tracker).track:51"}
```
-------------------

_**SN ID : 1HszZhvWCkXGscA4uV3eSTJwXFG1w8KjugLwFXGoqj1mRrYEzx**_

```
root@server030:/disk102/storj/logs/storagenode# grep "XLJMTPJAF66VICHFTKGSDALOSG6GCECPIZQT7XCMW6BA2FFEMKWA" ./old/server030* server030-v1.*
server030-v1.24.4-30030.log:2021-03-01T14:52:40.767+0100  	INFO    piecestore      upload started  {"Piece ID": "XLJMTPJAF66VICHFTKGSDALOSG6GCECPIZQT7XCMW6BA2FFEMKWA", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "PUT", "Available Space": 10716200335488}
server030-v1.24.4-30030.log:2021-03-01T14:52:44.169+0100  	INFO    piecestore      uploaded        {"Piece ID": "XLJMTPJAF66VICHFTKGSDALOSG6GCECPIZQT7XCMW6BA2FFEMKWA", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "PUT", "Size": 2212096}

server030-v1.30.2-30030.log:2021-06-01T17:31:26.188+0200        DEBUG   retain  About to move piece to trash    {"Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Piece ID": "XLJMTPJAF66VICHFTKGSDALOSG6GCECPIZQT7XCMW6BA2FFEMKWA", "Status": "enabled"}

server030-v1.30.2-30030.log:2021-06-02T22:54:54.721+0200        INFO    piecestore      download started        {"Piece ID": "XLJMTPJAF66VICHFTKGSDALOSG6GCECPIZQT7XCMW6BA2FFEMKWA", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "GET_AUDIT"}
server030-v1.30.2-30030.log:2021-06-02T22:54:54.721+0200        ERROR   piecestore      download failed {"Piece ID": "XLJMTPJAF66VICHFTKGSDALOSG6GCECPIZQT7XCMW6BA2FFEMKWA", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "GET_AUDIT", "error": "file does not exist", "errorVerbose": "file does not exist\n\tstorj.io/common/rpc/rpcstatus.Wrap:74\n\tstorj.io/storj/storagenode/piecestore.(*Endpoint).Download:508\n\tstorj.io/common/pb.DRPCPiecestoreDescription.Method.func2:217\n\tstorj.io/drpc/drpcmux.(*Mux).HandleRPC:33\n\tstorj.io/common/rpc/rpctracing.(*Handler).HandleRPC:58\n\tstorj.io/drpc/drpcserver.(*Server).handleRPC:111\n\tstorj.io/drpc/drpcserver.(*Server).ServeOne:62\n\tstorj.io/drpc/drpcserver.(*Server).Serve.func2:99\n\tstorj.io/drpc/drpcctx.(*Tracker).track:51"}

--

root@server030:/disk102/storj/logs/storagenode# grep "IAACJAHC44RH2E3GYOUIVYUJF7ZZSFTZC6H73UYHCDVCFYUPXRWQ" ./old/server030* server030-v1.*
server030-v1.22.2-30030.log:2021-02-23T15:00:22.727+0100  	INFO    piecestore      upload started  {"Piece ID": "IAACJAHC44RH2E3GYOUIVYUJF7ZZSFTZC6H73UYHCDVCFYUPXRWQ", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "PUT", "Available Space": 10811476922112}
server030-v1.22.2-30030.log:2021-02-23T15:00:24.708+0100  	INFO    piecestore      uploaded        {"Piece ID": "IAACJAHC44RH2E3GYOUIVYUJF7ZZSFTZC6H73UYHCDVCFYUPXRWQ", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "PUT", "Size": 2212096}

server030-v1.30.2-30030.log:2021-06-01T17:32:09.169+0200        DEBUG   retain  About to move piece to trash    {"Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Piece ID": "IAACJAHC44RH2E3GYOUIVYUJF7ZZSFTZC6H73UYHCDVCFYUPXRWQ", "Status": "enabled"}

server030-v1.30.2-30030.log:2021-06-02T23:24:08.724+0200        INFO    piecestore      download started        {"Piece ID": "IAACJAHC44RH2E3GYOUIVYUJF7ZZSFTZC6H73UYHCDVCFYUPXRWQ", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "GET_AUDIT"}
server030-v1.30.2-30030.log:2021-06-02T23:24:08.724+0200        ERROR   piecestore      download failed {"Piece ID": "IAACJAHC44RH2E3GYOUIVYUJF7ZZSFTZC6H73UYHCDVCFYUPXRWQ", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "GET_AUDIT", "error": "file does not exist", "errorVerbose": "file does not exist\n\tstorj.io/common/rpc/rpcstatus.Wrap:74\n\tstorj.io/storj/storagenode/piecestore.(*Endpoint).Download:508\n\tstorj.io/common/pb.DRPCPiecestoreDescription.Method.func2:217\n\tstorj.io/drpc/drpcmux.(*Mux).HandleRPC:33\n\tstorj.io/common/rpc/rpctracing.(*Handler).HandleRPC:58\n\tstorj.io/drpc/drpcserver.(*Server).handleRPC:111\n\tstorj.io/drpc/drpcserver.(*Server).ServeOne:62\n\tstorj.io/drpc/drpcserver.(*Server).Serve.func2:99\n\tstorj.io/drpc/drpcctx.(*Tracker).track:51"}

```
-------------------

_**SN ID : 1F1q1LLizj4aurSJV5Lv2sSC8ieAFpYgPrrrtG75bp6QLXhqHy**_

```
root@server030:/disk102/storj/logs/storagenode# grep "5EWJKVYSU6NPJG7KRM3PPDT73TGIPF4THRP3TDF26XAQHQDQUO5Q" ./old/server036*.log server036*.log
server036-v1.24.4-30036.log:2021-03-04T01:13:59.686+0100  	INFO    piecestore      upload started  {"Piece ID": "5EWJKVYSU6NPJG7KRM3PPDT73TGIPF4THRP3TDF26XAQHQDQUO5Q", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "PUT", "Available Space": 11486655561984}
server036-v1.24.4-30036.log:2021-03-04T01:14:03.097+0100  	INFO    piecestore      uploaded        {"Piece ID": "5EWJKVYSU6NPJG7KRM3PPDT73TGIPF4THRP3TDF26XAQHQDQUO5Q", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "PUT", "Size": 2212096}

server036-v1.30.2-30036.log:2021-06-01T17:12:36.540+0200        DEBUG   retain  About to move piece to trash    {"Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Piece ID": "5EWJKVYSU6NPJG7KRM3PPDT73TGIPF4THRP3TDF26XAQHQDQUO5Q", "Status": "enabled"}

server036-v1.30.2-30036.log:2021-06-02T23:46:54.143+0200        INFO    piecestore      download started        {"Piece ID": "5EWJKVYSU6NPJG7KRM3PPDT73TGIPF4THRP3TDF26XAQHQDQUO5Q", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "GET_AUDIT"}
server036-v1.30.2-30036.log:2021-06-02T23:46:54.143+0200        ERROR   piecestore      download failed {"Piece ID": "5EWJKVYSU6NPJG7KRM3PPDT73TGIPF4THRP3TDF26XAQHQDQUO5Q", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "GET_AUDIT", "error": "file does not exist", "errorVerbose": "file does not exist\n\tstorj.io/common/rpc/rpcstatus.Wrap:74\n\tstorj.io/storj/storagenode/piecestore.(*Endpoint).Download:508\n\tstorj.io/common/pb.DRPCPiecestoreDescription.Method.func2:217\n\tstorj.io/drpc/drpcmux.(*Mux).HandleRPC:33\n\tstorj.io/common/rpc/rpctracing.(*Handler).HandleRPC:58\n\tstorj.io/drpc/drpcserver.(*Server).handleRPC:111\n\tstorj.io/drpc/drpcserver.(*Server).ServeOne:62\n\tstorj.io/drpc/drpcserver.(*Server).Serve.func2:99\n\tstorj.io/drpc/drpcctx.(*Tracker).track:51"}
```
[Th3van.dk](http://th3van.dk)

**Comment by jtolio** [comment]
I know we haven't replied to this issue yet, but we've been working nonstop on figuring out what happened here since you filed it. Thank you very much for this report, it is much appreciated.

**Comment by Th3Van** [comment]
Seeing this again today on a few nodes 

```
root@server030:/disk102/storj/logs/storagenode# grep "6REJADTFNAWOW6F43H4JKED2TCBYWCHMUDQPI46LEMONEHOCVLUA" server004-v1.61.3-30004.log 

2022-08-21T22:51:38.083+0200    DEBUG   retain  About to move piece to trash    {"Process": "storagenode", "Satellite ID":"12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Piece ID": "6REJADTFNAWOW6F43H4JKED2TCBYWCHMUDQPI46LEMONEHOCVLUA", "Status": "enabled"}

2022-08-22T03:57:02.481+0200    INFO    piecestore      download started        {"Process": "storagenode", "Piece ID": "6REJADTFNAWOW6F43H4JKED2TCBYWCHMUDQPI46LEMONEHOCVLUA", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "GET_AUDIT"}
2022-08-22T03:57:02.545+0200    ERROR   piecestore      download failed {"Process": "storagenode", "Piece ID": "6REJADTFNAWOW6F43H4JKED2TCBYWCHMUDQPI46LEMONEHOCVLUA", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "GET_AUDIT", "error": "file does not exist", "errorVerbose": "file does not exist\n\tstorj.io/common/rpc/rpcstatus.Wrap:73\n\tstorj.io/storj/storagenode/piecestore.(*Endpoint).Download:547\n\tstorj.io/common/pb.DRPCPiecestoreDescription.Method.func2:228\n\tstorj.io/drpc/drpcmux.(*Mux).HandleRPC:33\n\tstorj.io/common/rpc/rpctracing.(*Handler).HandleRPC:61\n\tstorj.io/drpc/drpcserver.(*Server).handleRPC:122\n\tstorj.io/drpc/drpcserver.(*Server).ServeOne:66\n\tstorj.io/drpc/drpcserver.(*Server).Serve.func2:112\n\tstorj.io/drpc/drpcctx.(*Tracker).track:52"}
```
```
root@server030:/disk102/storj/logs/storagenode# grep "AYU7YR3H5SBESOWEEVI2RQ6SBKCQSZEB6QI5RVDK45RPYZGBFSWQ" server008-v1.61.3-30008.log 

2022-08-22T03:36:24.754+0200    DEBUG   retain  About to move piece to trash    {"Process": "storagenode", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Piece ID": "AYU7YR3H5SBESOWEEVI2RQ6SBKCQSZEB6QI5RVDK45RPYZGBFSWQ", "Status": "enabled"}

2022-08-22T14:42:27.547+0200    INFO    piecestore      download started        {"Process": "storagenode", "Piece ID": "AYU7YR3H5SBESOWEEVI2RQ6SBKCQSZEB6QI5RVDK45RPYZGBFSWQ", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "GET_AUDIT"}
2022-08-22T14:42:27.563+0200    ERROR   piecestore      download failed {"Process": "storagenode", "Piece ID": "AYU7YR3H5SBESOWEEVI2RQ6SBKCQSZEB6QI5RVDK45RPYZGBFSWQ", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "GET_AUDIT", "error": "file does not exist", "errorVerbose": "file does not exist\n\tstorj.io/common/rpc/rpcstatus.Wrap:73\n\tstorj.io/storj/storagenode/piecestore.(*Endpoint).Download:547\n\tstorj.io/common/pb.DRPCPiecestoreDescription.Method.func2:228\n\tstorj.io/drpc/drpcmux.(*Mux).HandleRPC:33\n\tstorj.io/common/rpc/rpctracing.(*Handler).HandleRPC:61\n\tstorj.io/drpc/drpcserver.(*Server).handleRPC:122\n\tstorj.io/drpc/drpcserver.(*Server).ServeOne:66\n\tstorj.io/drpc/drpcserver.(*Server).Serve.func2:112\n\tstorj.io/drpc/drpcctx.(*Tracker).track:52"}
```
```
root@server030:/disk102/storj/logs/storagenode# grep "DECUURRUQVSSJEEVMOEC4UXQQEUWGHBKMVSK44WQK6C3N5LITRMQ" server069-v1.61.3-30069.log 

2022-08-22T02:53:11.328+0200    DEBUG   retain  About to move piece to trash    {"Process": "storagenode", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Piece ID": "DECUURRUQVSSJEEVMOEC4UXQQEUWGHBKMVSK44WQK6C3N5LITRMQ", "Status": "enabled"}

2022-08-22T07:25:27.286+0200    INFO    piecestore      download started        {"Process": "storagenode", "Piece ID": "DECUURRUQVSSJEEVMOEC4UXQQEUWGHBKMVSK44WQK6C3N5LITRMQ", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "GET_AUDIT"}
2022-08-22T07:25:27.313+0200    ERROR   piecestore      download failed {"Process": "storagenode", "Piece ID": "DECUURRUQVSSJEEVMOEC4UXQQEUWGHBKMVSK44WQK6C3N5LITRMQ", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "GET_AUDIT", "error": "file does not exist", "errorVerbose": "file does not exist\n\tstorj.io/common/rpc/rpcstatus.Wrap:73\n\tstorj.io/storj/storagenode/piecestore.(*Endpoint).Download:547\n\tstorj.io/common/pb.DRPCPiecestoreDescription.Method.func2:228\n\tstorj.io/drpc/drpcmux.(*Mux).HandleRPC:33\n\tstorj.io/common/rpc/rpctracing.(*Handler).HandleRPC:61\n\tstorj.io/drpc/drpcserver.(*Server).handleRPC:122\n\tstorj.io/drpc/drpcserver.(*Server).ServeOne:66\n\tstorj.io/drpc/drpcserver.(*Server).Serve.func2:112\n\tstorj.io/drpc/drpcctx.(*Tracker).track:52"}
```
```
root@server030:/disk102/storj/logs/storagenode# grep "IHGSFDRZTXSM2LFJ3OZ73MT64Q7YOND6ECVLZGF4KP65DSFCCKDA" server097-v1.61.3-30097.log 

2022-08-21T02:45:37.299+0200    DEBUG   retain  About to move piece to trash    {"Process": "storagenode", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Piece ID": "IHGSFDRZTXSM2LFJ3OZ73MT64Q7YOND6ECVLZGF4KP65DSFCCKDA", "Status": "enabled"}

2022-08-22T08:50:49.582+0200    INFO    piecestore      download started        {"Process": "storagenode", "Piece ID": "IHGSFDRZTXSM2LFJ3OZ73MT64Q7YOND6ECVLZGF4KP65DSFCCKDA", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "GET_AUDIT"}
2022-08-22T08:50:49.606+0200    ERROR   piecestore      download failed {"Process": "storagenode", "Piece ID": "IHGSFDRZTXSM2LFJ3OZ73MT64Q7YOND6ECVLZGF4KP65DSFCCKDA", "Satellite ID": "12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S", "Action": "GET_AUDIT", "error": "file does not exist", "errorVerbose": "file does not exist\n\tstorj.io/common/rpc/rpcstatus.Wrap:73\n\tstorj.io/storj/storagenode/piecestore.(*Endpoint).Download:547\n\tstorj.io/common/pb.DRPCPiecestoreDescription.Method.func2:228\n\tstorj.io/drpc/drpcmux.(*Mux).HandleRPC:33\n\tstorj.io/common/rpc/rpctracing.(*Handler).HandleRPC:61\n\tstorj.io/drpc/drpcserver.(*Server).handleRPC:122\n\tstorj.io/drpc/drpcserver.(*Server).ServeOne:66\n\tstorj.io/drpc/drpcserver.(*Server).Serve.func2:112\n\tstorj.io/drpc/drpcctx.(*Tracker).track:52"}
```

Th3Van.dk

**Comment by littleskunk** [comment]
Sorry, we missed closing this ticket. The first time this happened it was a bug in the metainfo loop. This loop generated the bloom filters for all the storage nodes. We refactored the metainfo loop at some point. All of our tests looked great but in production we have very old segments that have a different format. The new metainfo loop bailed out with an error. The side effect of that error was that the storage nodes received incomplete bloom filters and started to delete pieces that they shouldn't delete.

Solution:
We fixed the metainfo loop to not error out in the first place and also make sure an error will abort garbage collection without sending unfinished bloom filters.
